geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureValidator.scala

```diff
@@ -8,168 +8,256 @@
 
 package org.locationtech.geomesa.convert
 
-import java.util.Date
+import java.util.{Date, Locale, ServiceLoader}
 
+import com.typesafe.scalalogging.LazyLogging
 import com.vividsolutions.jts.geom.Geometry
 import org.locationtech.geomesa.curve.BinnedTime
+import org.locationtech.geomesa.utils.geotools.wholeWorldEnvelope
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
 trait SimpleFeatureValidator {
-  def name: String
+
+  /**
+    * Initialize the validator with a simple feature type. Must be called before any calls to `validate`
+    *
+    * @param sft simple feature type
+    */
   def init(sft: SimpleFeatureType): Unit
-  def validate(sf: SimpleFeature): Boolean
-  def lastError: String
+
+  /**
+    * Validate a feature
+    *
+    * @param sf simple feature
+    * @return validation error message, or null if valid
+    */
+  def validate(sf: SimpleFeature): String
 }
 
-object SimpleFeatureValidator {
+object SimpleFeatureValidator extends LazyLogging {
 
-  val default: SimpleFeatureValidator = apply(Seq(HasDtgValidator.name, HasGeoValidator.name))
+  import scala.collection.JavaConverters._
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+  def default: SimpleFeatureValidator = apply(Seq(IndexValidator.name))
 
   def apply(names: Seq[String]): SimpleFeatureValidator = {
     val validators = names.map {
-      case "none"    => NoneValidator
-      case "has-geo" => new HasGeoValidator
-      case "has-dtg" => new HasDtgValidator
-      case "z-index" => new ZIndexValidator
-      case unk => throw new IllegalArgumentException(s"Unknown validator $unk")
+      case NoneValidator.name   => (NoneValidator, None)
```

Comment:
How does the NoneValidator interact with the other validators?  I.e., what happens with "['none', 'index']".

Reply:
it would just be an extra call that wouldn't do anything - since the none validator always passes, the end result would be equivalent to just the 'index' validator.